File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/Rate.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Entity
@Table(name = "rate")
public class Rate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "r_id")
    private Long rid;

    @Column(name = "r_value", nullable = false, columnDefinition = "DECIMAL(2,1)")
    private BigDecimal rvalue;

    @Column(name = "r_date", nullable = false)
    private java.sql.Timestamp rdate;

    @Column(name = "r_comment", length = 5120)
    private String rcomment;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "m_id")
    private Manga manga;
}

File: Frontend/manga-world/src/Model/Rate.ts
import Manga from "./Manga";
import User from "./User";

class Rate {
  rid: number;
  rvalue: number;
  rdate: Date;
  rcomment: string;
  userId: number;
  mid: number;
  user: User;
  manga: Manga;

  constructor(
    rid: number,
    rvalue: number,
    rdate: Date,
    rcomment: string,
    userId: number,
    mid: number,
    user: User,
    manga: Manga
  ) {
    this.rid = rid;
    this.rvalue = rvalue;
    this.rdate = rdate;
    this.rcomment = rcomment;
    this.userId = userId;
    this.mid = mid;
    this.user = user;
    this.manga = manga;
  }
}

export default Rate;
